### PR TITLE
Auro correct

### PR DESCRIPTION
**Pull request by valentk777**
hello. this is my pull request for autoCorrect. Now this is not working correctly and need to change dictionary words and add additional checking. On the other hand, I was made some additional changes such as negative number in receipt, percent of readied OCR and others. 

fell free to ask me questions if you want

### CODE REVIEW

File: GREEDY/App.config [hunk 1]
```diff
@@ -17,6 +17,7 @@
     <add key="MinPasswordLength" value="5" />
     <add key="MaxAnyInputLength" value="256" />
     <add key="AuthenticationSecret" value="db3OIsj+BXE9NZDy0t8W3TcNekrF+2d/1sFnWG4HnV8TZY30iTOdtVWJG8abWvB1GlOgJuQZdcF2Luqm/hccMw==" />
+    <add key="ClientSettingsProvider.ServiceUri" value="" />
```
Comment (MariusKurbakovas) [inline GREEDY/App.config:20]: What is this?

File: GREEDY/Brands.txt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿iphone
```
Comment (MariusKurbakovas) [inline GREEDY/Brands.txt:1]: Do we have to save all this stuff in .txt files (seems unprofessional) or is it just for now?

File: GREEDY/Extensions/GetReceiptExtensions.cs [hunk 3]
```diff
@@ -13,7 +13,18 @@ public static List<string> GetLinesOfText(this Emgu.CV.OCR.Tesseract source)
 
         public static List<string> GetLinesOfText(this Page source)
         {
-            return source.GetText().Split('\n').ToList();
+            var PercentageMatched = source.GetMeanConfidence();
+            if (PercentageMatched > 0.5)
```
Comment (MariusKurbakovas) [inline GREEDY/Extensions/GetReceiptExtensions.cs:17]: If you are not putting this value in config (which I would totally recommend) than atleast write a comment what it means. Values like these make the code unmaintainable (imagine that I only now joined this project and have no idea what is going on and need to change stuff like this)

File: GREEDY/GREEDY SERVER.csproj [hunk 4]
```diff
@@ -173,17 +178,19 @@
     <Compile Include="Migrations\Configuration.cs" />
     <Compile Include="ImagePreparation\IImageFormating.cs" />
     <Compile Include="Models\GraphData.cs" />
-    <Compile Include="ReceiptCreating\IItemCategorization.cs" />
+    <Compile Include="ReceiptCreatings\AutoCorrect.cs" />
+    <Compile Include="ReceiptCreatings\LevenshteinDistance.cs" />
+    <Compile Include="ReceiptCreatings\IItemCategorization.cs" />
     <Compile Include="ImagePreparation\ImageFormating.cs" />
-    <Compile Include="ReceiptCreating\IReceiptCreating.cs" />
+    <Compile Include="ReceiptCreatings\IReceiptCreating.cs" />
```
Comment (MariusKurbakovas) [inline GREEDY/GREEDY SERVER.csproj:185]: Can you give an explanation why you changed this folders name?

File: GREEDY/ReceiptCreatings/AutoCorrect.cs [hunk 5]
```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using GREEDY.Models;
+using NHunspell;
+using System.IO;
+using System.Linq;
+
+namespace GREEDY.ReceiptCreatings
+{
+    class AutoCorrect
```
Comment (MariusKurbakovas) [inline GREEDY/ReceiptCreatings/AutoCorrect.cs:10]: Again classes like these should be public not internal. https://stackoverflow.com/questions/4182015/internal-vs-public-in-c-sharp

File: GREEDY/ReceiptCreatings/AutoCorrect.cs [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+                lines = File.ReadAllLines("CustomWords-en_US.txt");
+                foreach (var line in lines)
+                {
+                    hunspell.Add(line);
+                }
+
+                //Main autoCorrect
+                string name = String.Empty;
+                string suggection = String.Empty;
```
Comment (MariusKurbakovas) [inline GREEDY/ReceiptCreatings/AutoCorrect.cs:33]: misstype in the name

File: GREEDY/ReceiptCreatings/LevenshteinDistance.cs [hunk 7]
```diff
@@ -0,0 +1,35 @@
+﻿using System;
+
+namespace GREEDY.ReceiptCreatings
+{
+    /// Contains approximate string matching
+    public class LevenshteinDistance
```
Comment (MariusKurbakovas) [inline GREEDY/ReceiptCreatings/LevenshteinDistance.cs:6]: You already created this class in the Extensions folder.

File: GREEDY/ReceiptCreatings/ReceiptCreating.cs [hunk 8]
```diff
@@ -33,12 +34,14 @@ public Receipt FullReceiptCreating(Bitmap image)
             var linesOfText = _ocr.ConvertImage(image);
             var date = GetDateForReceipt(linesOfText);
             var shop = GetShopFromData(linesOfText);
+            //var percentageMatched = float.Parse(linesOfText.Last(), CultureInfo.InvariantCulture.NumberFormat);
```
Comment (MariusKurbakovas) [inline GREEDY/ReceiptCreatings/ReceiptCreating.cs:37]: commented lines (atleast write above why is it commented and why it still may be needed)

File: GREEDY/Services/ReceiptService.cs [hunk 9]
```diff
@@ -32,6 +33,7 @@ public Receipt ProcessReceiptImage(Bitmap image)
                 image = _imageFormating.FormatImage(image);
                 var receipt = _receiptCreating.FullReceiptCreating(image);
                 receipt.ItemsList = _dataConverter.ReceiptToItemList(receipt);
+                //receipt.ItemsList = AutoCorrect.ItemsAutoCorrect(receipt.ItemsList);
```
Comment (MariusKurbakovas) [inline GREEDY/Services/ReceiptService.cs:36]: Again, write above why this is still left here (can even be in a TODO statement for the future)

**Review by MariusKurbakovas — COMMENTED** [review]
Requesting changes

**Review by MariusKurbakovas — COMMENTED** [review]
This branch and Brigita branch has Merge conflicts.

File: GREEDY/Services/ReceiptService.cs [hunk 10]
```diff
@@ -10,6 +10,7 @@ public class ReceiptService : IReceiptService
         private readonly IImageFormating _imageFormating;
         private readonly IReceiptCreating _receiptCreating;
         private readonly IDataConverter _dataConverter;
+        private static AutoCorrect AutoCorrect => new AutoCorrect();
```
Comment (AidasKil) [inline GREEDY/Services/ReceiptService.cs:13]: This is the same as writing private static AutoCorrect AutoCorrect {get {return new AutoCorrect();}
A new instance of will be created each time you reference this property.

File: GREEDY/ReceiptCreatings/AutoCorrect.cs [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+using GREEDY.Models;
+using NHunspell;
+using System.IO;
+using System.Linq;
+
+namespace GREEDY.ReceiptCreatings
+{
+    class AutoCorrect
+    {
+        private static LevenshteinDistance LevenshteinDistance => new LevenshteinDistance();
```
Comment (AidasKil) [inline GREEDY/ReceiptCreatings/AutoCorrect.cs:12]: Same thing as in ReceiptService, new instance will be created each time the property is referenced

File: GREEDY/OCRs/TesseractOcr.cs [hunk 12]
```diff
@@ -18,6 +18,7 @@ public List<string> ConvertImage(Bitmap image)
                 EngineMode.TesseractOnly
             );
             var page = _tesseract.Process(image);
+
```
Comment (AidasKil) [inline GREEDY/OCRs/TesseractOcr.cs:21]: don't commit stuff like this

**Comment by valentk777** [comment]
Thanks for comments. I will try to fix it in new branch

**Pull request closed without merging**